precompiles/common/types_test.go

```diff
@@ -5,13 +5,41 @@ import (
 
 	"cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/evmos/v18/cmd/config"
+	bankprecompile "github.com/evmos/evmos/v18/precompiles/bank"
+	"github.com/evmos/evmos/v18/precompiles/bech32"
 	"github.com/evmos/evmos/v18/precompiles/common"
+	distprecompile "github.com/evmos/evmos/v18/precompiles/distribution"
+	ics20precompile "github.com/evmos/evmos/v18/precompiles/ics20"
+	"github.com/evmos/evmos/v18/precompiles/p256"
+	stakingprecompile "github.com/evmos/evmos/v18/precompiles/staking"
+	vestingprecompile "github.com/evmos/evmos/v18/precompiles/vesting"
 	"github.com/evmos/evmos/v18/utils"
 	"github.com/stretchr/testify/require"
 )
 
 var largeAmt, _ = math.NewIntFromString("1000000000000000000000000000000000000000")
 
+func TestDefaultPrecompiles(t *testing.T) {
+	// We need to update the bech32 prefixes to use evmos1... instead of the default cosmos1...
+	cfg := sdk.GetConfig()
+	config.SetBech32Prefixes(cfg)
+
+	require.Equal(t, 7, len(common.DefaultPrecompilesBech32),
+		"expected different number of default precompiles",
+	)
+	require.Equal(t, []string{
+		sdk.AccAddress(ethcommon.HexToAddress(p256.PrecompileAddress).Bytes()).String(),
+		sdk.AccAddress(ethcommon.HexToAddress(bech32.PrecompileAddress).Bytes()).String(),
+		sdk.AccAddress(ethcommon.HexToAddress(stakingprecompile.PrecompileAddress).Bytes()).String(),
+		sdk.AccAddress(ethcommon.HexToAddress(distprecompile.PrecompileAddress).Bytes()).String(),
+		sdk.AccAddress(ethcommon.HexToAddress(ics20precompile.PrecompileAddress).Bytes()).String(),
+		sdk.AccAddress(ethcommon.HexToAddress(vestingprecompile.PrecompileAddress).Bytes()).String(),
+		sdk.AccAddress(ethcommon.HexToAddress(bankprecompile.PrecompileAddress).Bytes()).String(),
+	}, common.DefaultPrecompilesBech32)
```

Comment:
Ideally, we shouldn't have the same address hardcoded twice. There should be only one source of truth. Can we refactor this use smth like `sdk.AccAddress(ethcommon.HexToAddress(bankprecompile.PrecompileAddress).Bytes()).String()` in the definition of the `DefaultPrecompilesBech32 instead of hardcoding these?

Reply:
I agree that would be the best case, because as it is right now this is also pretty error-prone, because we can just forget to add the address of a new precompile to this list and it will be possible to send funds to it. 

However this would require some bigger refactors it looks like and this PR was just a quick safety measure to at least have a check in place which wasn't there before. The problem I'm seeing why it's not so straight forward is, that the `BlockedAddrs()` method where this slice of Bech32 addresses is passed is necessary to pass into the creation of the bank keeper. Then subsequently, the bank keeper needs to be passed into the creation of the EVM keeper, so we can't just use the `GetAvailablePrecompileAddrs()` method to return the actual addresses.

I can adjust it, so that instead of the hardcoded values we are actually computing the addresses on the spot when running the `BlockedAddrs()` method, but it will still be manually filled with entries, so that won't be solved for now 👀